pkg/cloud/utils.go

```diff
@@ -135,10 +135,12 @@ func (r *ResourceID) String() string {
 }
 
 // apiGroupRegex is used to extract the API Group out of a Resource URL.
-// This regex expects API Group to be followed by <ver>/projects/ path
-// segments. Unfortunately it cannot predict what comes before the API
+// This regex expects API Group to be followed ine one of 2 patterns:
+// <ver>/projects/ path or legacy one <api_group>.googleapis.com/<ver>/projects/.
+// Unfortunately it cannot predict what comes before the API
 // group since that is configurable via SetAPIDomain.
-var apiGroupRegex = regexp.MustCompile(`([a-z]*)\/(alpha|beta|v1|v1alpha1|v1beta1)/projects`)
+// legacyApiGroupRegex is used to extract API Group from legacy path in format
+var apiGroupRegex = regexp.MustCompile(`([a-z]*)(\.googleapis\.com)*\/(alpha|beta|v1|v1alpha1|v1beta1)/projects`)
```

Comment:
should we have `(\.googleapis\.com)*` <== star, seems like we want to have `(\.googleapis\.com)?`

Reply:
start is because we have 1 regex for 2 rules legacy and new one. In new one googleapis is before compute (the group API we are looking for) so we just ignores it.

Reply:
done